### PR TITLE
Updates error handling to use typed errors and masks

### PR DESCRIPTION
**Pull request by JosephSalisbury**
Towards https://github.com/giantswarm/architect/issues/34

### CODE REVIEW

File: template/template_test.go [hunk 1]
```diff
@@ -32,18 +32,18 @@ func TestTemplateHelmChart(t *testing.T) {
...
 			},
 			check: func(fs afero.Fs, helmPath string) error {
 				fileInfos, err := afero.ReadDir(fs, helmPath)
 				if err != nil {
-					return err
+					return microerror.MaskAny(err)
 				}
 				if len(fileInfos) != 0 {
-					return fmt.Errorf("multiple files found in helm directory")
+					return multipleHelmChartsError
```
Comment (rossf7) [inline template/template_test.go]: Is this the right error? I got confused with the multiple files error?

File: utils/error.go [hunk 2]
```diff
@@ -0,0 +1,19 @@
+package utils
+
+import (
+	"github.com/juju/errgo"
+)
+
+var sourceIsNotDirectoryError = errgo.New("source is not a directory")
+
+// IsSourceIsNotDirectory asserts sourceIsNotDirectoryError.
```
Comment (rossf7) [inline utils/error.go]: Nitpick but this method name reads strangely to me. Although I can see it follows the convention.

File: template/template_test.go (hunk 1 above)
Comment (kopiczko) [inline template/template_test.go]: return microerror.MaskAny(multipleHelmChartsError)

File: template/error.go [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+}
+
+var resourceNotFoundError = errgo.New("did not find the required resource")
+
+// IsResourceNotFound asserts resourceNotFoundError.
+func IsResourceNotFound(err error) bool {
+	return errgo.Cause(err) == resourceNotFoundError
+}
+
+var incorrectTemplatingError = errgo.New("did not find required field from template")
```
Comment (kopiczko) [inline template/error.go]: Error messages doesn't conform "our standards". Please consult [this guide](https://github.com/giantswarm/giantswarm/wiki/go-style-guide) for further information.

Bureaucracy. Guide was written today, so I'm open to discussion, but I saw @xh3b4sd complaining about it. Anf I personally see the benefits.

It should be `var incorrectTemplatingError = errgo.New("incorrect templating")`

**Comment by JosephSalisbury** [comment]
@kopiczko let's go, i don't wanna spend too much more time on this :D

**Comment by JosephSalisbury** [comment]
or @rossf7 for the re-approve? i think pawel is afk

**Pull request merged**